[Bug] Potential NullPointerException in ShareActivity

**Description:**
We inspected the latest release version of Syncthing (1.20.4 from F-droid) using an Android application testing tool under development, and found some points that potentially cause crash.

In **exported** Activity `ShareActivity`, function `ShareActivity.onCreate()`, [line 122](https://github.com/syncthing/syncthing-android/blob/fb5ba8b73e65bfe29cf9da98acf290069a3b79a5/app/src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java#L122) and [line 126](https://github.com/syncthing/syncthing-android/blob/fb5ba8b73e65bfe29cf9da98acf290069a3b79a5/app/src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java#L126) in ShareActivity.java, unsafe `String.equals()` usage may produce NullPointerException.

```java
    if (getIntent().getAction().equals(Intent.ACTION_SEND)) {  // L122: NPE when action is null
    ...
    } else if (getIntent().getAction().equals(Intent.ACTION_SEND_MULTIPLE)) {  // L126: NPE when action is null
    ...
```

Therefore, the fix goes with recommended `String.equals()` usage:

```java
    if (Intent.ACTION_SEND.equals(getIntent().getAction())) {
    ...
    } else if (Intent.ACTION_SEND_MULTIPLE.equals(getIntent().getAction())) {
    ...
```

**Stacktrace:**

```log
E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.nutomic.syncthingandroid/com.nutomic.syncthingandroid.activities.ShareActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2698)
E AndroidRuntime: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2759)
E AndroidRuntime: 	at android.app.ActivityThread.-wrap12(ActivityThread.java)
E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1482)
E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6190)
E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:892)
E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:782)
E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.equals(java.lang.Object)' on a null object reference
E AndroidRuntime: 	at com.nutomic.syncthingandroid.activities.ShareActivity.onCreate(ShareActivity.java:122)
E AndroidRuntime: 	at android.app.Activity.performCreate(Activity.java:6701)
E AndroidRuntime: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
E AndroidRuntime: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2651)
```

**Version Info:**
 - App Version: 1.20.4
 - Android Version: Android 7.1.2
 - Device Model: Emulator

--- a/app/src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java
+++ b/app/src/main/java/com/nutomic/syncthingandroid/activities/ShareActivity.java
@@ -119,11 +119,11 @@ public class ShareActivity extends StateDialogActivity
 
         // TODO: add support for EXTRA_TEXT (notes, memos sharing)
         ArrayList<Uri> extrasToCopy = new ArrayList<>();
-        if (getIntent().getAction().equals(Intent.ACTION_SEND)) {
+        if (Intent.ACTION_SEND.equals(getIntent().getAction())) {
             Uri uri = getIntent().getParcelableExtra(Intent.EXTRA_STREAM);
             if (uri != null)
                 extrasToCopy.add(uri);
-        } else if (getIntent().getAction().equals(Intent.ACTION_SEND_MULTIPLE)) {
+        } else if (Intent.ACTION_SEND_MULTIPLE.equals(getIntent().getAction())) {
             ArrayList<Uri> extras = getIntent().getParcelableArrayListExtra(Intent.EXTRA_STREAM);
             if (extras != null)
                 extrasToCopy = extras;

